### PR TITLE
Custom endpoint

### PR DESCRIPTION
**Pull request by kevinmusker**
These changes allow a custom endpoint (e.g. an S3 compatible interface to a locally hosted service such as http://docs.ceph.com/docs/master/radosgw/s3/) to be specified in the thumbor config.

It used to be possible to specific a custom endpoint in the configuration of previous versions of boto, but not in boto3.

### CODE REVIEW

**Comment by kevinmusker** [comment]
I'll take a look at fixing these tests.

**Comment by kevinmusker** [comment]
I fixed a bug when constructing a Bucket for the storage backends (I had only considered the loader).

File: tc_aws/aws/session.py [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
+__all__ = ['get_session']
 
-session = None
+sessions = [None, None]
 
-
-def get_session():
-    global session
+def get_session(endpoint=None):
+    session = sessions[bool(endpoint)]
```
Comment (kevinmusker) [inline tc_aws/aws/session.py]: Hmm, you're right. Its only possible to use standard S3, or a single custom endpoint, not both at the same time.

I'll make the change on Monday.

**Comment by kevinmusker** [comment]
@vmalaga, do you have the latest version?

e443d09a640fd53a7b8d3542f0fced889f2ad144 adds the missing argument to the Bucket construction.

File: tc_aws/aws/session.py [hunk 2]
```diff
@@ -1,11 +1,20 @@
 import botocore.session
+from botocore.utils import fix_s3_host
 
+__all__ = ['get_session']
 
 session = None
 
-
-def get_session():
+def get_session(custom_endpoint=None):
```
Comment (Bladrak) [inline tc_aws/aws/session.py]: This edit isn't needed anymore :)

Comment (Bladrak) [inline tc_aws/aws/session.py]: Oh my bad, misread

File: tc_aws/aws/bucket.py [hunk 3]
```diff
@@ -35,9 +38,10 @@ def get(self, path, callback=None):
         :param string path: Path or 'key' to retrieve AWS object
         :param callable callback: Callback function for once the retrieval is done
         """
-        my_session = session_handler.get_session()
+        my_session = session_handler.get_session(self._endpoint)
```
Comment (Bladrak) [inline tc_aws/aws/bucket.py]: I'd rather see a boolean value than a string one here, if you can make the change

**Comment by kevinmusker** [comment]
@Bladrak change made as requested - apologies for the delay.

**Comment by Bladrak** [comment]
Alright, seems good to me :) I'll let this open for a day or two in case anyone else would like to review it, and then merge it & release it.

**Comment by Bladrak** [comment]
This has been released on pypi as version 6.0.2

**Comment by vmalaga** [comment]
Hi @Bladrak sorry for the question, but i am not sure. 
This mean that the modifications @kevinmusker made to support a custom endpoint  I can find it with
```
pip install tc_aws==6.0.2
```

**Comment by Bladrak** [comment]
Yes, should be fine with that :)

**Pull request merged**